azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java

```diff
@@ -23,11 +35,102 @@
      * The maven cli argument for set the active subscription by id or name
      */
     @Parameter(property = "subscription")
-    private String subscription;
+    private String subscriptionId;
 
     @Override
-    public void execute() throws MojoExecutionException {
-        throw new NotImplementedException("Not implemented.");
+    public void execute() throws MojoExecutionException, MojoFailureException {
```

Comment:
Shall we split this method? Seems it contains a bit much operations,

Reply:
there is a TODO comment.